src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java

```diff
@@ -2,17 +2,51 @@
 
 import com.kodilla.ecommercee.dto.ProductDto;
 import com.kodilla.ecommercee.entity.Product;
+import com.kodilla.ecommercee.entity.ProductGroup;
+import com.kodilla.ecommercee.service.ProductGroupService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Component
 public class ProductMapper {
 
-    public Product mapToProduct(ProductDto productDto) {
-        return new Product();
+    @Autowired
+    private ProductGroupService productGroupService;
+
+    public Product mapToProduct(final ProductDto productDto) {
```

Comment:
Całość działa, ale trzeba to rozbić:
Tu powinno się znaleźć to co jest niżej
ProductGroup productGroup = productGroupService.getGroupOrException(productDto.getGroupId())